src/plugins/saved_objects_management/public/plugin.ts

```diff
@@ -18,24 +18,44 @@
  */
 
 import { i18n } from '@kbn/i18n';
+import { CoreSetup, CoreStart, Plugin } from 'src/core/public';
+import { HomePublicPluginSetup, FeatureCatalogueCategory } from '../../home/public';
 import {
-  FeatureCatalogueCategory,
-  HomePublicPluginSetup,
-} from '../../../../../../../plugins/home/public';
-import { SavedObjectsManagementActionRegistry } from './saved_objects_management_action_registry';
+  SavedObjectsManagementActionRegistry,
+  ISavedObjectsManagementActionRegistry,
+} from './services';
 
-interface SetupDependencies {
+export interface SavedObjectsManagementPluginSetup {
+  actionRegistry: ISavedObjectsManagementActionRegistry;
```

Comment:
Should `actionRegistry.has` and `actionRegistry.getAll` be part of the public contract?

I wonder if we shouldn't just expose the register methods from the top-level like 
```
export interface SavedObjectsManagementPluginSetup {
  registerAction: (action: SavedObjectsManagementAction) => ...
}
```

Reply:
- `actionRegistry.has` seems unused, and I was going to remove it during next stage (after another check) from the action registry.

- `actionRegistry.getAll` is used in `src/legacy/core_plugins/kibana/public/management/sections/objects/components/objects_table/components/table/table.js` via `npSetup`. The legacy call will be adapted in the next PR to use the action registry from the new NP plugin once migrated

- The next stage of the migration will introduce additional APIs to `SavedObjectsManagementPluginSetup`, that's why I wanted to isolate `actions` with a nested service. I could change the API to something like `SavedObjectsManagementPluginSetup.actions.register` and `SavedObjectsManagementPluginSetup.actions.getAll` though.

Reply:
yeah since this is just a part of a larger refactor it makes sense to merge this and update the API in a later PR.